### PR TITLE
Add recipe to user list

### PR DESCRIPTION
**Pull request by dmachineni**
- created functionality to add recipes (from the search page and other lists) to previously created user lists
- implemented logic that ensures users are logged in before rendering this feature

### CODE REVIEW

File: backend/models/recipes.js [hunk 1]
```diff
@@ -92,11 +92,22 @@ class Recipes {
...
+                  let videoUrl = await parseObj.get("videoUrl")
+                  let ingredientsInfo = await parseObj.get("ingredientsInfo")
+
                   let obj = {
-                    "id":id,
-                    "name":name,
-                    "thumbnail_url":pic,
-                    "description":description
+                    id:id,
+                    name:name,
```
Comment (ksio2124fb) [inline backend/models/recipes.js]: i think you could put 
`let obj {id, name, thumbnail_url: pic....}`

File: ui/src/components/App/App.jsx [hunk 2]
```diff
@@ -69,7 +68,11 @@ export default function App() {
   }
 
   const handleOnSearchChange = (searchInput) => {
-    axios.get(`http://localhost:3001/${searchInput}`)
+    if(searchInput === "") {
+      return
+    }
+
+    axios.get(`http://localhost:3001/search/${searchInput}`)
```
Comment (ksio2124fb) [inline ui/src/components/App/App.jsx:75]: i always get excited when i see string templating.

**Review by ksio2124fb — APPROVED** [review]

File: ui/src/components/App/App.jsx [hunk 3]
```diff
@@ -116,11 +116,11 @@ export default function App() {
                   handleAddRecipe={handleAddRecipe} setChosenRecipe={setChosenRecipe}/>
               </div>
             }/>
-            <Route path="/search" element={
+            <Route path="/search" onClick={()=>alert("boo")} element={
```
Comment (ksio2124fb) [inline ui/src/components/App/App.jsx]: intentional boo alert?

File: ui/src/components/ListDetails/ListDetails.jsx [hunk 4]
```diff
@@ -10,7 +10,6 @@ export default function ListDetails(props) {
     if (props.pic === "") {
         let len = props.subCatRecipes.length
         let randomNum = Math.floor(Math.random() * len);
-        console.log(props.subCatRecipes[randomNum].thumbnailUrl)
         props.subCatRecipes[randomNum].thumbnailUrl === undefined ?
```
Comment (ksio2124fb) [inline ui/src/components/ListDetails/ListDetails.jsx]: curious if we could write this instead:
`
const randomRecipe = props.subCatRecipes[randomNum]`
`profilePic = randomRecipe.thumbnailUrl === undefined ? randomRecipe.thumbail_url : randomRecipe.thumbnailUrl`

**Review by ksio2124fb — APPROVED** [review]

File: ui/src/components/ListDetails/ListDetails.jsx (hunk 4 above)
Comment (dmachineni) [reply ui/src/components/ListDetails/ListDetails.jsx]: just changed it!

File: backend/models/recipes.js (hunk 1 above)
Comment (dmachineni) [reply backend/models/recipes.js]: just changed it!

**Pull request merged**